cpp/modules/deck.gl/core/src/lib/deck.h

```diff
@@ -71,7 +71,8 @@ class Deck : public Component {
   int width{100};   // Dummy value, ensure something is visible if user forgets to set window size
   int height{100};  // Dummy value, ensure something is visible if user forgets to set window size
 
-  std::shared_ptr<LayerContext> context{new LayerContext{this}};
+  std::shared_ptr<lumagl::AnimationLoop> animationLoop{std::make_shared<lumagl::GLFWAnimationLoop>()};
+  std::shared_ptr<LayerContext> context{new LayerContext{this, *this->animationLoop->device.get()}};
```

Comment:
Maybe keep an internal `_device` on animation loop that is a shared_ptr, and a public device that is just wgpu::Device so we can hide this ugliness completely.